### PR TITLE
Add ssl options to webclient in oidc-client

### LINKED ISSUE #14480: OIDC Client > Cannot overwrite properties
**Describe the bug**
Question about override properties for Quarkus apps:

The thing is that when deploying these apps into OpenShift, the apps are failing because:

```
__  ____  __  _____   ___  __ ____  ______
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/
 -/ // / // / __ |/ , / ,< / // /\ \
--\___\\____// |//|//|_|\____/___/
2021-01-21 11:07:36,224 ERROR [io.qua.run.Application] (main) Failed to start application (with profile prod): java.net.ConnectException: Connection refused
    at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
    at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:779)
    at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:330)
    at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)
    at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:702)
    at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:650)
    at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:576)
    at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
    at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
    at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
    at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
    at java.base/java.lang.Thread.run(Thread.java:834)
```

I've checked that the problem is that the OIDC client properties are not being replaced correctly. These are the environment properties that I'm passing:

`QUARKUS_OIDC_AUTH_SERVER_URL` (this property is for the existing OIDC extension and it's working fine: it's mapped to `quarkus.oidc.auth-server-url`)
`QUARKUS_OIDC_CLIENT_AUTH_SERVER_URL` (this should be mapped to `quarkus.oidc-client.auth-server-url`)

Same with the OIDC Client tenants. 

**Expected behavior**
The new extensions should work as the existing OIDC extension being able to overwrite properties when deploying apps into the cloud via environment properties.

**Actual behavior**
Application is failing to start.

**Environment (please complete the following information):**
 - Quarkus version or git rev: 1.11.0.Final and master

### PR DESCRIPTION
**Pull request by sd-f**
Fixes #14480

### CODE REVIEW

File: extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java [hunk 1]
```diff
@@ -147,6 +149,31 @@ public OidcClient apply(String tokenRequestUri) {
         });
     }
 
+    private static void setHttpsClientOptions(URI authServerUri, WebClientOptions options) {
```
Comment (sberyozkin) [inline extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java:152]: @sd-f can you please rename it to `setWebClientOptions` ? It is not only https properties that are set there, right now `trustAll` is dealt in `OidcCommonUtils`...

Comment (sd-f) [reply extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java]: @sberyozkin done :)

**Comment by sberyozkin** [comment]
@sd-f I think we also need to update the oidc client test to use HTTPS; would you like to try it ? 
Apply the same changes as [here](https://github.com/quarkusio/quarkus/blob/master/integration-tests/oidc/src/test/java/io/quarkus/it/keycloak/KeycloakRealmResourceManager.java#L24), [here](https://github.com/quarkusio/quarkus/blob/master/integration-tests/oidc/src/test/java/io/quarkus/it/keycloak/KeycloakRealmResourceManager.java#L24), [here](https://github.com/quarkusio/quarkus/blob/master/integration-tests/oidc/src/test/java/io/quarkus/it/keycloak/KeycloakRealmResourceManager.java#L24), [here](https://github.com/quarkusio/quarkus/blob/master/integration-tests/oidc/src/test/java/io/quarkus/it/keycloak/KeycloakRealmResourceManager.java#L24) (the same property is referenced twice), [here](https://github.com/quarkusio/quarkus/blob/master/integration-tests/oidc/src/test/java/io/quarkus/it/keycloak/KeycloakRealmResourceManager.java#L24), [here](https://github.com/quarkusio/quarkus/blob/master/integration-tests/oidc/src/test/java/io/quarkus/it/keycloak/KeycloakRealmResourceManager.java#L24)(as well as the https port a few lines below) and finally [here](https://github.com/quarkusio/quarkus/blob/0f33c23aac875295693d00cef073adabbff8d780/integration-tests/oidc/src/main/resources/application.properties#L2) (see also a trust all property) to `integration-tests/oidc-client`.

Give it a try please if you don't mind, I can help as well

**Comment by sd-f** [comment]
> 
> Give it a try please if you don't mind, I can help as well

I'll try

**Comment by sberyozkin** [comment]
@sd-f thanks

**Comment by sd-f** [comment]
> @sd-f thanks

I am not sure why but it worked without `quarkus.oidc.tls.verification=none` or `trust-all`
I added
```
    static {
        RestAssured.useRelaxedHTTPSValidation();
    }
```
and this seems to be enouth. 

I also added the https endpoint to the keycloak docker container in maven.

And finally I hope my formatter worked xD

**Comment by sberyozkin** [comment]
@sd-f Great,  Re the import order, that should be the same after building the source (I've tested it in some other file), did you rebuild `oidc-client/runtime` :-) ?
Please also squash the commits

**Comment by sd-f** [comment]
> @sd-f Great, Re the import order, that should be the same after building the source (I've tested it in some other file), did you rebuild `oidc-client/runtime` :-) ?
> Please also squash the commits

Thank your for your help, i followed the contribution guide and imported "import order config" for intellij after that the order was way different. mvn checks and verify is not complaining (formatter, ismimport plugin)

**Comment by sberyozkin** [comment]
@sd-f Np, thanks; `Initial JDK 11 Build` is failing right now with:
```
Error:  Failed to execute goal net.revelc.code:impsort-maven-plugin:1.4.1:check (check-imports) on project quarkus-oidc-client: Imports are not sorted in /home/runner/work/quarkus/quarkus/extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal net.revelc.code:impsort-maven-plugin:1.4.1:check (check-imports) on project quarkus-oidc-client: Imports are not sorted in /home/runner/work/quarkus/quarkus/extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java
```

What happens if you just do a clean rebuild on the cmd line ? I think it should restore them; if it does not - simply restore the original order for now - may be the import order config is not effective for IntelliJ ... CC @gsmet

**Comment by sd-f** [comment]
> @sd-f Np, thanks; `Initial JDK 11 Build` is failing right now with:
> 
> ```
> Error:  Failed to execute goal net.revelc.code:impsort-maven-plugin:1.4.1:check (check-imports) on project quarkus-oidc-client: Imports are not sorted in /home/runner/work/quarkus/quarkus/extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java -> [Help 1]
> org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal net.revelc.code:impsort-maven-plugin:1.4.1:check (check-imports) on project quarkus-oidc-client: Imports are not sorted in /home/runner/work/quarkus/quarkus/extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java
> ```
> 
> What happens if you just do a clean rebuild on the cmd line ? I think it should restore them; if it does not - simply restore the original order for now - may be the import order config is not effective for IntelliJ ... CC @gsmet

Thanks, already fixed, I did the imports by hand. impsort:check is successful and I did a fresh build on command line before commit :)

**Comment by sd-f** [comment]
> @sd-f Np, thanks; `Initial JDK 11 Build` is failing right now with:
> 
> ```
> Error:  Failed to execute goal net.revelc.code:impsort-maven-plugin:1.4.1:check (check-imports) on project quarkus-oidc-client: Imports are not sorted in /home/runner/work/quarkus/quarkus/extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java -> [Help 1]
> org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal net.revelc.code:impsort-maven-plugin:1.4.1:check (check-imports) on project quarkus-oidc-client: Imports are not sorted in /home/runner/work/quarkus/quarkus/extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java
> ```
> 
> What happens if you just do a clean rebuild on the cmd line ? I think it should restore them; if it does not - simply restore the original order for now - may be the import order config is not effective for IntelliJ ... CC @gsmet

**Comment by sberyozkin** [comment]
@sd-f cool; thanks sorting it out :-)

**Comment by sberyozkin** [comment]
@sd-f OK, the builds are good, so lets merge it. 
Can you please give me a favour and check sometime later how would using `.getAbs()` (vs `.get()`) change the picture. @vsevel uses `.getAbs()` in his PR.
The question I have, what is cheaper. If set here the default host and port then may be it is more efficient to use only the path component of the auth server url and also token url ? At the moment we set the default host/port - but also use absolute URIs with the `.get()` and `post()`. It feels like we can further optimize something :-). I'll open an issue and CC you

Thanks for your contribution

**Review by sberyozkin — APPROVED** [review]

**Comment by rsvoboda** [comment]
@gsmet fyi, this should get into 1.11.1.

**Pull request merged**